File: BE/src/main/java/com/ssafy/e106/domain/subscription/dto/request/SubscriptionCreateRequest.java
package com.ssafy.e106.domain.subscription.dto.request;

import java.time.LocalDate;

import jakarta.validation.constraints.NotBlank;

public record SubscriptionCreateRequest(
    @NotBlank(message = "subscriptionType은 필수입니다.")
    String subscriptionType,
    Long serviceId,
    Long servicePlanId,
    String bundleCode,
    LocalDate nextBillingDate) {
}

File: FE/app/src/main/java/com/ssafy/e106/data/dto/subscription/SubscriptionCreateRequest.kt
package com.ssafy.e106.data.dto.subscription

import kotlinx.serialization.Serializable

@Serializable
data class SubscriptionCreateRequest(
    val subscriptionType: String,
    val serviceId: Long? = null,
    val servicePlanId: Long? = null,
    val bundleCode: String? = null,
    val nextBillingDate: String? = null,
)
